cluster.go

```diff
@@ -241,7 +254,8 @@ func (c *clusterClient) _refresh() (err error) {
 
 	c.mu.RLock()
 	for addr, cc := range c.conns {
-		if fresh, ok := conns[addr]; ok {
+		fresh, ok := conns[addr]
+		if ok && cc.replica == fresh.replica {
```

Comment:
```suggestion
		if ok && (cc.replica == fresh.replica || c.opt.SendToReplicas == nil) {
```

Reply:
We can still reuse the conn if c.opt.SendToReplicas is not configured.

Reply:
[87e2b39](https://github.com/redis/rueidis/pull/430/commits/87e2b39036c247c1106d95207277848d3c7b1f2c)

Reply:
Hi @proost, thank you very much! It looks great!